### PR TITLE
Add guide to troubleshoot Ballerina runtime

### PR DESCRIPTION
**Pull request by Nadeeshan96**
$title

## Checklist

- [x] **Page addition**
  - [x] Add `permalink` to pages.

- [ ] **Page removal**
  - [ ] Remove entry from corresponding left nav YAML file.
  - [ ] Add `redirect_from` on the alternative page.
  - [ ] If no alternative page, add redirection on the `redirections.js ` file.

- [ ] **Page rename**
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

- [ ] **Page restrcuture**
  - [ ] Add `permalink` to pages.
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

### CODE REVIEW

File: learn/guides/troubleshoot-ballerina-runtime.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+While the program is running, obtain its PID.
+```
+$ jps
+3408 Main
+28851 Jps
+28845 $_init
+```
+Here for this program we get the PID as 28845, because `$_init` is the entry `class` filename of the Ballerina program. 
+To get the strand dump, send the `SIGTRAP` signal to that process. You can use the CLI command
```
Comment (warunalakshitha) [inline learn/guides/troubleshoot-ballerina-runtime.md:87]:
```suggestion
To get the strand dump, send the `SIGTRAP` signal to that process. You can use the CLI command.
```

File: learn/guides/troubleshoot-ballerina-runtime.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+```
+$ kill -SIGTRAP 28845
+```
+or
+```
+$ kill -5 28845
+```
+
+Then the strand dump at that moment will be emitted to the standard output stream of the Ballerina program.
```
Comment (warunalakshitha) [inline learn/guides/troubleshoot-ballerina-runtime.md:96]:
```suggestion
Then the dump of the runtime strands will be emitted to the standard output stream of the Ballerina program. 
```

File: learn/guides/troubleshoot-ballerina-runtime.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+  - /swan-lake/learn/troubleshoot-ballerina-runtime
+  - /learn/troubleshooting-ballerina-runtime/
+  - /learn/troubleshooting-ballerina-runtime
+  - /learn/ballerina-strand-dump/
+  - /learn/ballerina-strand-dump
+---
+
+Ballerina runtime can have unexpected behaviors due to user code errors, bugs or issues with the running environment. 
+These will result in memory leaks, CPU spinning, runtime hangs, performance degradation or crashes with various errors. 
+Ballerina provides a tool to dump the status of currently running strands.
```
Comment (warunalakshitha) [inline learn/guides/troubleshoot-ballerina-runtime.md:19]: @praneesha Can u review this introduction section.

File: learn/guides/troubleshoot-ballerina-runtime.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+                at      ballerina.lang.runtime.0.0.0:sleep(runtime.bal:61)
+                        demo.strandDump.0.1.0:$lambda$_0(main.bal:13)
+ 
+        strand 5 "receiver" [demo.strandDump.0:addnum][3] [BLOCKED ON WORKER MESSAGE RECEIVE]:
+                at      demo.strandDump.0.1.0:$lambda$_1(main.bal:18)
+ 
+===========================================
+```
+
+### Format and available details
```
Comment (HindujaB) [inline learn/guides/troubleshoot-ballerina-runtime.md:122]: Shall we have this?
```suggestion
### Output format and available details
```

**Comment by Nadeeshan96** [comment]
Closing this because a new PR will be sent to the `release-2201.2.0-new` branch.

**Pull request closed without merging**